core/src/main/java/org/apache/accumulo/core/client/impl/ScannerOptions.java

```diff
@@ -244,4 +247,20 @@ public void setBatchTimeout(long timeout, TimeUnit timeUnit) {
   public long getBatchTimeout(TimeUnit timeUnit) {
     return timeUnit.convert(batchTimeOut, TimeUnit.MILLISECONDS);
   }
+
+  @Override
+  public void setContext(String context) {
+    this.context = context;
```

Comment:
>   However if we do not accept null (users must call clearContext insead)

IMO, we either need to have clear javadoc on setContext about what a null context means or just fail on null. I'd prefer the latter since there's an explicit clearContext method. That gives us more flexibility in the future to hide behind the interface.